classes/user/Collector.php

```diff
@@ -654,9 +656,16 @@ protected function buildReviewerStatistics(Builder $query): self
         }
 
         $disableSharedReviewerStatistics = (bool) Application::get()->getRequest()->getSite()->getData('disableSharedReviewerStatistics');
-        $dateDiff = fn (string $dateA, string $dateB): string => DB::connection() instanceof MySqlConnection
-            ? "DATEDIFF({$dateA}, {$dateB})"
-            : "DATE_PART('day', {$dateA} - {$dateB})";
+
+        $dateDiff = fn (string $dateA, string $dateB): string => match (true) {
+            DB::connection() instanceof MySqlConnection,
+            DB::connection() instanceof MariaDbConnection
+                => "DATEDIFF({$dateA}, {$dateB})",
+            DB::connection() instanceof PostgresConnection
+                => "DATE_PART('day', {$dateA} - {$dateB})",
+            default
```

Comment:
`match` will already take care of throwing an exception when there's no matched condition and no default is specified; I'd suggest letting that take care of it. https://php.watch/versions/8.0/match-expression#must-match